ophyd/v2/core.py

```diff
@@ -402,96 +405,150 @@ def monitor_reading_value(self, callback: ReadingValueCallback[T]) -> Monitor:
 }
 
 
-class SimMonitor(Generic[T]):
+class SimMonitor(Monitor):
     """Handle that is returned when monitoring a Signal in sim mode"""
 
-    def __init__(
-        self, callback: ReadingValueCallback[T], listeners: List[SimMonitor[T]]
-    ):
-        self.callback = callback
-        self._listeners = listeners
-        self._listeners.append(self)
+    task: Awaitable[asyncio.Task[None]]
+
+    def __init__(self, task: asyncio.Task[None]):
+        self.task = cast(Awaitable[asyncio.Task[None]], task)
 
     def close(self):
-        self._listeners.remove(self)
+        pass
```

Comment:
We need to make sure that `close` actually stops updates.

However, thinking about this, a `SignalBackend` shouldn't have to handle multiple subscribers, as there is only one, the `_SignalCache`.

How about changing `monitor_reading_value(cb: Callback[...]) -> Monitor` to `set_callback(cb: Optional[Callback[...]]) -> None` which tells the backend of a possible single subscriber to be notified when the reading and value changes. It should still have the same behaviour of immediately calling it on the current value, but means there is no need to keep a list of listeners in the backend.

Reply:
So should the stuff using `Monitor`, mainly SimSignalBackend, now use `_SignalCache` instead?